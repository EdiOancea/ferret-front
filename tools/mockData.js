const company = {
  id: 1,
  name: 'Name',
  address: 'Address',
  business: 'Business',
  rating: 5,
  manager: 'Manager',
  active: true,
};

export { company };
