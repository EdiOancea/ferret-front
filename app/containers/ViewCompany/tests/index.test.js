import React from 'react';
import { render } from 'enzyme';
import { fromJS } from 'immutable';
import {
  ViewCompany,
  mapStateToProps,
  mapDispatchToProps,
} from 'containers/ViewCompany';
import { company } from '../../../../tools/mockData';

describe('<ViewCompany />', () => {
  it('should get the error if view company failed', () => {
    const getCompanyState = fromJS({
      errorMessage: 'This is an error',
    });
    const initialState = fromJS({
      getCompany: getCompanyState,
    });
    expect(mapStateToProps(initialState).errorMessage).toEqual(
      'This is an error',
    );
  });

  it('should view company when page is loaded', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchCompany('1');
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: 'GET_COMPANY',
      id: '1',
    });
  });

  it('should render the company item correctly', () => {
    const fetchCompany = jest.fn();
    const tree = render(
      <ViewCompany id="1" company={company} fetchCompany={fetchCompany} />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render nothing when company does not exist', () => {
    const fetchCompany = jest.fn();
    const tree = render(
      <ViewCompany id={NaN} company={{}} fetchCompany={fetchCompany} />,
    );
    expect(tree).toMatchSnapshot();
  });
});
