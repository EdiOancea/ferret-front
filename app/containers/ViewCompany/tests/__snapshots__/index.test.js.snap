// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewCompany /> should render nothing when company does not exist 1`] = `
<div
  class="makeStyles-root-1"
>
  <div>
    <div
      class="makeStyles-field-2"
    >
      Name
    </div>
    <div
      class="makeStyles-field-2"
    />
  </div>
  <div>
    <div
      class="makeStyles-field-2"
    >
      Address
    </div>
    <div
      class="makeStyles-field-2"
    />
  </div>
  <div>
    <div
      class="makeStyles-field-2"
    >
      Business
    </div>
    <div
      class="makeStyles-field-2"
    />
  </div>
  <div>
    <div
      class="makeStyles-field-2"
    >
      Rating
    </div>
    <div
      class="makeStyles-field-2"
    >
      <div
        class="Rating-root-1"
      >
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div>
    <div
      class="makeStyles-field-2"
    >
      Manager
    </div>
    <div
      class="makeStyles-field-2"
    />
  </div>
  <div>
    <div
      class="makeStyles-field-2"
    >
      Active
    </div>
    <div
      class="makeStyles-field-2"
    >
      True
    </div>
  </div>
</div>
`;

exports[`<ViewCompany /> should render the company item correctly 1`] = `
<div
  class="makeStyles-root-1"
>
  <div>
    <div
      class="makeStyles-field-2"
    >
      Name
    </div>
    <div
      class="makeStyles-field-2"
    >
      Name
    </div>
  </div>
  <div>
    <div
      class="makeStyles-field-2"
    >
      Address
    </div>
    <div
      class="makeStyles-field-2"
    >
      Address
    </div>
  </div>
  <div>
    <div
      class="makeStyles-field-2"
    >
      Business
    </div>
    <div
      class="makeStyles-field-2"
    >
      Business
    </div>
  </div>
  <div>
    <div
      class="makeStyles-field-2"
    >
      Rating
    </div>
    <div
      class="makeStyles-field-2"
    >
      <div
        class="Rating-root-1"
      >
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root-12 MuiIconButton-root-6 Rating-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-11"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-15 Rating-icon-3"
              color="#e0e0e0"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div>
    <div
      class="makeStyles-field-2"
    >
      Manager
    </div>
    <div
      class="makeStyles-field-2"
    />
  </div>
  <div>
    <div
      class="makeStyles-field-2"
    >
      Active
    </div>
    <div
      class="makeStyles-field-2"
    >
      True
    </div>
  </div>
</div>
`;
